decidim-assemblies/lib/decidim/assemblies/engine.rb

```diff
@@ -12,11 +12,21 @@ class Engine < ::Rails::Engine
       isolate_namespace Decidim::Assemblies
 
       routes do
-        resources :assemblies, only: [:index, :show], path: "assemblies" do
+        get "assemblies/:assembly_id", to: redirect { |params, _request|
+          assembly = Decidim::Assembly.find(params[:assembly_id])
+          assembly ? "/assemblies/#{assembly.slug}" : "/404"
```

Comment:
@mrcasals Isn't this line dead code now? `find` either rises or returns a record. I don't think we need the manual "/404" redirection anymore.

Reply:
EDIT: Not the whole line, just the "/404" branch.

Reply:
Rechecking this, the `find` method makes the constraint to fail, but I'm getting an authorization error instead of a nice 404 error:

![](https://i.imgur.com/kYEuymb.png)

I think I will go back to what we used to have, and then investigate again.

Reply:
Definitely unexpected... :thinking:

> I think I will go back to what we used to have, and then investigate again.

Agree on the approach :+1:

Reply:
Wait, I misschecked it. This is why I'm getting with the current code:

![](https://i.imgur.com/hkrUdzO.png)

The previous error appears after accessing a with a wrong slug: `http://localhost:3000/assemblies/does-not-exist` raises an authorization error instead of a 404.

I'll look into it, since it seems to be caused by this PR.

Reply:
OK, last commit should fix it.

Reply:
:+1: If you want to test this stuff, I added some support for it a while ago. It's the `a 404 page` shared example.